src/LibraryManager/LibraryNaming/VersionedLibraryNamingScheme.cs

```diff
@@ -26,15 +28,16 @@ class VersionedLibraryNamingScheme : ILibraryNamingScheme
                 return (name, version);
             }
 
-            int indexOfAt = libraryId.LastIndexOf(Separator);
+            int indexOfLastAt = libraryId.LastIndexOf(Separator);
```

Comment:
with the else-if suggested above, you can make this `libraryId.Substring(1).LastIndexOf(Separator) + 1;`.  This removes the need to further calculate `indexOfAtToSplitNameAndVersion`.  (`indexOfLastAt` is now 0 for a non-match, but you're already checking for > 0 below).

Actually, I think it can even further simplify below, back to a nice and easy:

```
if (indexOfAt > 0)
 {
    name = libraryId.Substring(0, indexOfAt);
    version = libraryId.SubString(indexOfAt + 1);
}
else
{
    name = libraryId;
}
```

I don't think the `libraryId.TrimEnd().Length - 1` check is relevant for a non-whitespace separator, and the else block replaces the `name = indexOfLastAt > 0 ? libraryId.TrimEnd('@') : libraryId;` line.  The interesting test case for this one is "@@@", which should return ("@@", "") I believe - trim the final @ as the separator, but TrimEnd('@') would remove all of them.

Reply:
Resolved in commit 2